source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+    }
+
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::parse_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial conditions");
+      {
+        prm.enter_subsection("Adiabatic boundary");
+        {
+          isotherm_file_name = prm.get("Isotherm depth filename");
+          isotherm_temperature = prm.get_double("Isotherm temperature");
+          surface_temperature  = prm.get_double("Surface temperature");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      std::ifstream input1(isotherm_file_name.c_str());
+      AssertThrow (input1.is_open(),
+                   ExcMessage (std::string("Can't read from file <") + isotherm_file_name + ">"));
+
+      /**
+       * Reading the data file.
+       */
+      while (true)
+        {
+          double latitude_iso, longitude_iso, depth_iso;
+          input1 >> latitude_iso >>longitude_iso >> depth_iso;
+          if (input1.eof())
+            break;
+          latitudes_iso.push_back(latitude_iso);
+          longitudes_iso.push_back(longitude_iso);
+          depths_iso.push_back(depth_iso);
+        }
+      /**
+       * Find first 2 numbers that are different to use in
+       * calculating half the difference between each position as delta.
+       */
+
+      if (std::fabs(latitudes_iso[0] - latitudes_iso[1]) > 1e-9)
+        {
+          /**
+           * Calculate delta as half the distance between points.
+          */
+          delta = std::fabs((0.5)*(latitudes_iso[0] - latitudes_iso[1]));
+        }
+      else
+        {
+          /**
+           * Calculate delta as half the distance between points.
+          */
+          delta = std::fabs((0.5)*(longitudes_iso[0] - longitudes_iso[1]));
+        }
+
+    }
+
+  }
+}
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    ASPECT_REGISTER_INITIAL_CONDITIONS(AdiabaticBoundary,
+                                       "adiabatic boundary",
+                                       "An initial temperature condition that allows for discretizing "
+                                       "the model domain into two layers separated by a user-defined "
+                                       "isotherm boundary using a table look-up approach. The user includes an "
+                                       "input ascii data file that is formatted as 3 columns of 'latitude', "
+                                       "'longitude', and 'depth', where 'depth' is in kilometers and "
+                                       "represents that depth of an initial temperature of 1673.15 K (by default). "
+                                       "The temperature is defined from the surface (273.15 K) to the isotherm "
+                                       "as a linear gradient. Below the isotherm the temperature increases "
+                                       "adiabatically (0.0005 K per meter). This initial temperature condition "
```

Comment:
I am somewhat hesitant calling this adiabatically, because in fact it is linear as in the lithosphere above (just with a different coefficient). Could you add "approximately adiabatically by constant  0.0005 K per meter. Also this number could make a great input parameter (optional).